target/ppc/fpu_helper.c

```diff
@@ -414,6 +414,49 @@ void helper_store_fpscr(CPUPPCState *env, uint64_t val, uint32_t nibbles)
     ppc_store_fpscr(env, val);
 }
 
+void helper_fpscr_check_status(CPUPPCState *env)
```

Comment:
helper_float_check_status shouldn't on itself update fp_status based on fpscr, as one of the things it does is update fpscr based on fp_status.

The problem with updating fp_status based on fpscr is that some exception bits present in fpscr are not present in fp_status, and usually this is dealt with looking at the last operation and its operand values (for example, the fpscr bit `FP_VXISI` is set when there's an add/sub operation, the fp_status bit `float_flag_invalid` is set and both values are infinities being subtracted), so helper_float_check_status doesn't check for those.  
Other problem is dealing with the bit VXSOFT, as it is an exception bit that should throw an exception, but can only be changed by `mtfsf`, `mtfsfi` and `mtfsb1`, so there's no fp_status value that should throw its exception.

As for when to call the new function vs when to call the old function is based on what are you checking, fp_status is updated when a floating point operation occurs, so it should be called after fp operations when you intend to update FPSCR and throw the appropriate exception. The new function should be called when the FPSCR is updated directly and you intend to throw an exception based on its new value.

Reply:
Thanks for the information.

My suggestions then would be to add a brief comment to the new helper explaining when it should be called (to update fp_status and raise exceptions after directly updating FPSCR?), and possibly rename the helper to ``helper_check_fpscr`` or ``helper_float_check_fpscr`` (to avoid confusion between checking fp_status and fpscr).